test/common/upstream/resource_manager_impl_test.cc

```diff
@@ -12,26 +12,35 @@ TEST(ResourceManagerImplTest, RuntimeResourceManager) {
   ResourceManagerImpl resource_manager(
       runtime, "circuit_breakers.runtime_resource_manager_test.default.", 0, 0, 0, 0);
 
-  EXPECT_CALL(runtime.snapshot_,
-              getInteger("circuit_breakers.runtime_resource_manager_test.default.max_connections",
-                         0U)).WillOnce(Return(1U));
+  EXPECT_CALL(
+      runtime.snapshot_,
+      getInteger("circuit_breakers.runtime_resource_manager_test.default.max_connections", 0U))
+      .Times(2)
+      .WillRepeatedly(Return(1U));
   EXPECT_EQ(1U, resource_manager.connections().max());
+  EXPECT_TRUE(resource_manager.connections().canCreate());
```

Comment:
while we are here, i'd add some tests where canCreate returns false for a given resource (connections or pending requests, etc)